Source/SourceKittenFramework/SourceKitObject.swift

```diff
@@ -81,9 +89,8 @@ public struct SourceKitObject {
     ///   - key: The key to associate with value. If key already exists in the dictionary, 
     ///     value replaces the existing associated value. If key isn't already a key of the dictionary
     public func updateValue(_ value: SourceKitObjectConvertible, forKey key: UID) {
-        precondition(sourcekitdObject != nil)
-        precondition(value.sourcekitdObject != nil)
-        sourcekitd_request_dictionary_set_value(sourcekitdObject!, key.uid, value.sourcekitdObject!)
+        precondition(value.sourceKitObject != nil)
+        sourcekitd_request_dictionary_set_value(sourcekitdObject, key.sourcekitdUID, value.sourceKitObject!.sourcekitdObject)
```

Comment:
This line here worries me slightly because I think we may need to retain `value.sourceKitObject`. Theoretically we could make `children` mutable and just append it to the array. We would be leaking the old value, but only for the lifetime of the dictionary.

My other suggestion in the PR description would also resolve this potential issue.

Reply:
Even with this concern, this is still leaps and bounds better than what we had before. I'm not too concerned in practice.

Reply:
Let me do a little testing. I'm worried that we will set the value and then free it, which could result in memory corruption.

I'll also add those changelogs.

Reply:
There are plenty of unit tests that are stressing this function, but all with `Int64`s. Since they are a c primitive I figured there might be room for this to cause issues in the future. I decided to add this sourceKitObject to the children array, then it's lifecycle will be tied to it's parent. Seems like the safest option.

Reply:
Sounds good. Thanks for being thorough!